Update dashboard to pass survey ID to revised reporting endpoint

AdamHawtin opened the pull request:
# Motivation and Context
The reporting endpoint is changing to require a survey ID, and the reporting service no longer distinguishes between EQ and SEFT reports.

# What has changed
- Updated reporting API endpoint to pass survey ID but not collection instrument type
- Updated HTML with survey ID 
- Update JS to pass survey ID from HTML to reporting API gateway endpoint
- Updated tests
- Updated mock endpoints
- Fixed travis build

# How to test?
Run the service against this branch of rm-reporting: https://github.com/ONSdigital/rm-reporting/pull/15
It should still load the dashboard and show correct figures.

# Links
https://github.com/ONSdigital/rm-reporting/tree/88-dont-count-dummy-sample-units
https://trello.com/c/kp7foNVs/87-bug-accounts-enrolled-and-pending-counts-dont-filter-by-survey-id
https://trello.com/c/Kl95brGU/88-dont-count-dummy-sample-units
https://github.com/ONSdigital/rm-reporting/pull/15

AdamHawtin commented:
![](https://github.trello.services/images/mini-trello-icon.png) [BUG: Accounts enrolled and pending counts don't filter by survey id](https://trello.com/c/kp7foNVs/87-bug-accounts-enrolled-and-pending-counts-dont-filter-by-survey-id)

AdamHawtin commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Don't count dummy sample units](https://trello.com/c/Kl95brGU/88-dont-count-dummy-sample-units)

MebinAbraham reviewed (approved):
👍 Tested against RM reporting branch.

AdamHawtin committed:
Updated reporting API endpoint to pass survey ID but not collection instrument type
Updated HTML with survey ID
Update JS to pass survey ID from HTML to reporting API gateway endpoint
Updated tests
Updated mock endpoints

ryangrundy7 commented on app/api/reporting.py:
```diff
@@ -26,9 +29,13 @@ def reporting_details(collection_instrument_type, collex_id):
         abort(404)
 
-    return report
+    if collection_instrument_type == 'seft':
```
Why aren't we doing the same for eQ?

AdamHawtin replied:
The figures the reporting service returns are already all in the format we use for EQ, it's only SEFT that needs any further calculation done at the moment.

AdamHawtin committed:
Validate survey ID format, update tests

AdamHawtin committed:
Fix travis build to build python and JS in separate sub-jobs
